src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurCashAdvancePdpFeedFileServiceImpl.java

```diff
@@ -51,32 +53,73 @@ public boolean createPdpFeedFileForValidatedDetailFileLines(ConcurRequestExtract
         return pdpFileSuccessfullyCreated;
     }
 
-    private PdpFeedFileBaseEntry buildPdpFeedBaseEntry(ConcurRequestExtractFile requestExtractFile) {
+    private PdpFeedFileBaseEntry buildPdpFeedBaseEntry(ConcurRequestExtractFile requestExtractFile, ConcurRequestExtractBatchReportData reportData) {
         int totalPdpDetailRecordsCount = 0;
         KualiDecimal totalPdpDetailRecordsAmount = KualiDecimal.ZERO;
+        int totalClonedCashAdvanceLinesCount = 0;
+        KualiDecimal totalClonedCashAdvanceLinesAmount = KualiDecimal.ZERO;
+        int totalCashAdvancesUseInExpenseReportCount = 0;
+        KualiDecimal totalCashAdvancesUseInExpenseReportAmount= KualiDecimal.ZERO;
+        int totalDuplicateCashAdvanceCount = 0;
+        KualiDecimal totalDuplicateCashAdvanceAmount = KualiDecimal.ZERO;
+        int totalNonCashAdvanceRecordsCount = 0;
+        KualiDecimal totalNonCashAdvanceRecordsAmount = KualiDecimal.ZERO;
+        int totalCashAdvancesNotProcessedDueToValidationErrorsCount = 0;
+        KualiDecimal totalCashAdvancesNotProcessedDueToValidationErrorsAmount = KualiDecimal.ZERO;
+        int totalProcessingAttemptedLineCount = 0;
+        KualiDecimal totalProcessingAttemptedLineAmount = KualiDecimal.ZERO;
```

Comment:
When I first coded the report data processing into this function, I had it coded as you suggested and it did not seem read as clearly which values were being updated and why.  If you and Daniela feel strongly about this method using the report data object values directly, I can refactor the code to remove the "updateReportData..." methods and just do that summing directly against the reportData attributes.  Let me know your preference.